File: CloudFlow/src/ro/cloudSoft/cloudDoc/presentation/client/shared/model/content/GetAvailableReplacementProfilesInWhichRequesterIsReplacementRequestModel.java
package ro.cloudSoft.cloudDoc.presentation.client.shared.model.content;

import java.io.Serializable;
import java.util.Collection;
import java.util.Date;

public class GetAvailableReplacementProfilesInWhichRequesterIsReplacementRequestModel implements Serializable {

	private static final long serialVersionUID = 1L;
	
	private Long idForRequestingReplacementProfile;
	private Collection<Long> idsForRequesterUserProfiles;
	private Date startDate;
	private Date endDate;
	
	public Long getIdForRequestingReplacementProfile() {
		return idForRequestingReplacementProfile;
	}
	public void setIdForRequestingReplacementProfile(Long idForRequestingReplacementProfile) {
		this.idForRequestingReplacementProfile = idForRequestingReplacementProfile;
	}
	public Collection<Long> getIdsForRequesterUserProfiles() {
		return idsForRequesterUserProfiles;
	}
	public void setIdsForRequesterUserProfiles(Collection<Long> idsForRequesterUserProfiles) {
		this.idsForRequesterUserProfiles = idsForRequesterUserProfiles;
	}
	public Date getStartDate() {
		return startDate;
	}
	public void setStartDate(Date startDate) {
		this.startDate = startDate;
	}
	public Date getEndDate() {
		return endDate;
	}
	public void setEndDate(Date endDate) {
		this.endDate = endDate;
	}
}

File: CloudFlow/FrontEnd-App/src/app/shared/model/content/get-available-replacement-profiles-in-which-requester-is-replacement-request.model.ts
import { JsonObject, JsonProperty } from "json2typescript";
import { JsonDateConverter } from "../../json-mapper";

@JsonObject
export class GetAvailableReplacementProfilesInWhichRequesterIsReplacementRequestModel {

	@JsonProperty("idForRequestingReplacementProfile", Number)
	public idForRequestingReplacementProfile: number = null;

	@JsonProperty("idsForRequesterUserProfiles", [Number])
	public idsForRequesterUserProfiles: number[] = null;

	@JsonProperty("startDate", JsonDateConverter)
	public startDate: Date = null;

	@JsonProperty("endDate", JsonDateConverter)
	public endDate: Date = null;
}